src/ngraph/shape.hpp

```diff
@@ -25,46 +25,80 @@
 namespace ngraph
 {
     /// \brief Shape for a tensor.
-    class Shape : public std::vector<size_t>
+    class Shape
```

Comment:
This class is missing traits. Standard `algorithms` use traits to operate on containers.
```
using value_type = std::size_t;
using allocator_type = typename std::vector<std::size_t>::allocator_type;
using size_type = std::size_t;
using difference_type = std::ptrdiff_t
using reference  = value_type&;
using const_reference = const value_type&;
using pointer = std::allocator_traits<allocator_type>::pointer;
using const_pointer = std::allocator_traits<allocator_type>::const_pointer;
using iterator = typename std::vector<value_type>::iterator;
using const_iterator = const iterator;
using reverse_iterator = std::reverse_iterator<iterator>;
using const_reverse_iterator = std::reverse_iterator<const_iterator>;
```

Reply:
This is not intended to be an stl container, just a Shape. If we don't need something then why go to the effort of implementing and testing it?